test/SpecHelpers.hs

```diff
@@ -165,6 +163,12 @@ testEvaluating
 
 type Val term = Value term Precise
 
+evaluateProject :: (HasPrelude lang, SLanguage lang) => TaskSession -> Proxy lang -> [FilePath] -> IO (TestEvaluatingState term (TestEvaluatingResult term))
+evaluateProject session proxy = case parserForLanguage analysisParsers lang of
+  Just (SomeParser parser) -> unsafeCoerce . testEvaluating <=< evaluateProject' session proxy parser
+  _                        -> error $ "analysis not supported for " <> show lang
+  where lang = reflect proxy
```

Comment:
We want to select an abstract parser, but the various language-specific analysis tests _also_ want to be able to compare and equate the values and errors that result, which means that we can’t abstract the term types existentially (because there’s no guarantee that two existentially-quantified values inhabit the same type without e.g. `Typeable` shenanigans) or by coercing to `Any` (because `Any` doesn’t have an `Eq` instance).

I didn’t care to go to all the effort of the `Typeable` approach, so we just `unsafeCoerce`. This is safe, so long as the term type you select corresponds to the term type for the parser for the language; which it does because we only use this in the handful of analysis tests which are going to go away anyway.

Reply:
I had never seen `Any` before. Horrifying.

But this is fine, I think, especially given that this is going away. I would ask for `Typeable` if it were something that would last longer.